src/types/__test__/2/campaign.test.ts

```diff
@@ -0,0 +1,436 @@
+import type { Connections, User } from 'definitions'
+
+import { v4 } from 'uuid'
+
+import {
+  IMAGE_ASSET_TYPE,
+  LANGUAGE,
+  NODE_TYPES,
+  CAMPAIGN_STATE,
+  CAMPAIGN_USER_STATE,
+} from 'common/enums'
+import { CampaignService, SystemService, AtomService } from 'connectors'
+import { toGlobalId } from 'common/utils'
+
+import { genConnections, closeConnections, testClient } from '../utils'
+
+let connections: Connections
+let campaignService: CampaignService
+let systemService: SystemService
+let atomService: AtomService
+
+beforeAll(async () => {
+  connections = await genConnections()
+  campaignService = new CampaignService(connections)
+  systemService = new SystemService(connections)
+  atomService = new AtomService(connections)
+}, 30000)
+
+afterAll(async () => {
+  await closeConnections(connections)
+})
+
+describe('create or update wrting challenges', () => {
+  const PUT_WRITING_CHALLENGE = /* GraphQL */ `
+    mutation ($input: PutWritingChallengeInput!) {
+      putWritingChallenge(input: $input) {
+        id
+        shortHash
+        name
+        description
+        cover
+        link
+        applicationPeriod {
+          start
+          end
+        }
+        writingPeriod {
+          start
+          end
+        }
+        stages {
+          name
+          period {
+            start
+            end
+          }
+        }
+        state
+      }
+    }
+  `
+  const translations = [
+    {
+      text: 'test' + LANGUAGE.zh_hant,
+      language: LANGUAGE.zh_hant,
+    },
+    { text: 'test' + LANGUAGE.zh_hans, language: LANGUAGE.zh_hans },
+    { text: 'test' + LANGUAGE.en, language: LANGUAGE.en },
+  ]
+  const name = translations
+  const description = translations
+  let admin: User
+  let normalUser: User
+  let cover: string
+  const link = 'https://test.com'
+  const applicationPeriod = {
+    start: new Date('2024-01-01'),
+    end: new Date('2024-01-02'),
+  }
+  const writingPeriod = {
+    start: new Date('2024-01-03'),
+    end: new Date('2024-01-04'),
+  }
+  const stages = [
+    {
+      name: translations,
```

Comment:
should use different translations as stage name 

ref: comment `src/queries/campaign/stage/name.ts#L9`